screenpy/resolutions/custom_matchers/is_in_bounds.py

```diff
@@ -62,16 +74,20 @@ def describe_mismatch(self, item: float, mismatch_description: Description) -> N
         )
 
 
-def is_in_bounds(*bounds: int | (float | str)) -> IsInBounds:
+def is_in_bounds(*bounds: int | float | str) -> IsInBounds:
     """Matches a number that falls within the bounds."""
     lower_comparator = operator.le
     upper_comparator = operator.le
     if len(bounds) == 1:
         bounding_string = str(bounds[0])
-        pattern = (
-            r"^(?P<lower>[\[\(]?)"
-            r"(?P<minorant>\d+).*?(?P<majorant>\d+)"
-            r"(?P<upper>[\]\)]?)$"
+        pattern = re.compile(
+            r"^"  # start of line
+            r"(?P<lower>[\[(]?)"  # [ or (
+            r"(?P<minorant>-?(\d+\.?\d*)|(\.\d+))"  # minorant (int or float)
+            r"[\-, ]+?"  # any number of comma, dash and space
+            r"(?P<majorant>-?(\d+\.?\d*)|(\.\d+))"  # majorant (int or float)
+            r"(?P<upper>[])]?)"  # ) or ]
+            r"$"  # end of line
```

Comment:
Splitting this up to add explanatory comments was a good idea. 👍

Reply:
Right?  As someone who has a hard time trying to read complex regex, this helps so much.